lib/core/hooks.js

```diff
@@ -34,24 +37,14 @@ module.exports = function (env_config, callback) {
     return false;
   };
 
-  // todo: refactor run() to use this
-  var getHooksFromPlugins = function(type, name) {
-    return _.compact(_.map(env_config.plugins, function(plugin) {
-      if(plugin.hooks[type] && plugin.hooks[type][name]) {
-        return plugin.hooks[type][name];
-      }
-    }));
-  };
-
   var runAsyncEvery = function(type, name, args, callback) {
-    var runHook = function(hook, callback) {
-      args.push(callback);
```

Comment:
While this very well may an issue, the versions I've been testing with have identical hooks.js files, yet the newer server version leaks.  :( I'll still run tests against the newer build for reference though and let you know.